component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -164,6 +169,30 @@ public boolean isValidAssertion(SignedJWT signedJWT) throws OAuthClientAuthnExce
                 issuedTime = issuedAtTime.getTime();
             }
 
+            //   Obtain the signing algorithm used to sign the JWT in the request.
+            String requestSigningAlgorithm = signedJWT.getHeader().getAlgorithm().getName();
+            if (!isValidSignatureAlgorithm(requestSigningAlgorithm, consumerKey)) {
+                throw new OAuthClientAuthnException("Signature algorithm used in the request is invalid.",
+                        OAuth2ErrorCodes.INVALID_CLIENT);
+            }
+
+            /* For FAPI compliant applications the allowed JWT signing algorithm should be configured at the
+               application creation and only PS256 and ES256 algorithms are allowed. Therefore these will be checked
+               against the signing algorithm used to sign the JWT in the request.
+               https://openid.net/specs/openid-financial-api-part-2-1_0.html#algorithm-considerations */
+            if (OAuth2Util.isFapiConformantApp(consumerKey)) {
+                //   Mandating FAPI specified JWT signing algorithms.
+                List<String> fapiAllowedSigningAlgorithms = IdentityUtil
+                        .getPropertyAsList(FAPI_SIGNATURE_ALG_CONFIGURATION);
+                if (!fapiAllowedSigningAlgorithms.contains(requestSigningAlgorithm)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("FAPI unsupported signing algorithm " + requestSigningAlgorithm +
+                                " is used to sign the JWT.");
+                    }
+                    return false;
```

Comment:
Why is it that in one place we are throwing an exception[1] and here we are sending false when the used sig alg is not accepted?

[1] - https://github.com/wso2-extensions/identity-oauth-addons/pull/108/files#diff-c1fefc87f2cf0485b2f5e352367c515cb1036459a5232698108ff48d4d50a5e1R175

Reply:
Fixed in e9d0e7eeb2260413659c89164eed65584a7d4278